code/scripting/api/objs/subsystem.cpp

```diff
@@ -26,12 +26,12 @@ ship_subsys_h::ship_subsys_h() : object_h() {
 ship_subsys_h::ship_subsys_h(object* objp_in, ship_subsys* sub) : object_h(objp_in) {
 	ss = sub;
 }
-bool ship_subsys_h::isSubsystemValid() const { return object_h::IsValid() && objp->type == OBJ_SHIP && ss != nullptr; }
+bool ship_subsys_h::isSubsystemValid() const { return object_h::isValid() && objp->type == OBJ_SHIP && ss != nullptr; }
```

Comment:
So, I wonder whether it's possible to make this an override of isValid instead of a different function to allow this change to also be usable for subsystems.
Then again, if a significant portion of the codebase relies on being able to distinguish whether a subsystem is invalid but its parent object is not, that might be hard.

Reply:
Why would it be an override of `object_h`'s `isValid`?  The `ship_subsys_h` handle isn't a subclass of `object_h`.

And yes, subsystems can be valid if the parent ship isn't valid.  An example would be looking up a nonexistent subsystem on a valid ship.

Reply:
Unfortunately, ship_subsys_h _is_ a subclass of object_h, for whatever godforsaken reason.
ship_subsys isn't of object, but their lua handles are related classes

Reply:
Yeah, you're right, the subsystem handle should also use `isValid`.  I'll make the change and roll it into the earlier refactor commit.